### ISSUE TITLE
Slider/RangeSlider: TalkBack screen reader on Android fails to announce output text content

### ISSUE THREAD

**Issue by majornista** [adobe/react-spectrum#1400]
# 🐛 Bug Report 

TalkBack screen reader on Android fails to announce output text content.

## 🤔 Expected Behavior

Screen reader should always announce the value of the output element within the Slider/RangeSlider grouping.

## 😯 Current Behavior

Using TalkBack on Android, the Slider/RangeSlider grouping is only announced using the label referenced using `aria-labelledby`, ignoring the text content of the output tag. 

## 💁 Possible Solution

Add a self-referential IDREF to the `output` element's `aria-labelledby` attribute.

## 🔦 Context

Identified as an issue in Dec 17 release testing.

## 💻 Code Sample

```jsx
...
    outputProps: {
      htmlFor: state.values.map((_, index) => getSliderThumbId(state, index)).join(' '),
      id: outputId,
      'aria-labelledby': `${labelProps.id} ${outputId}`,
      'aria-live': 'off'
    }
...
```

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 2.21.0-beta.0
| Browser          | Chrome
| Operating System | Android with TalkBack enabled

## 🧢 Your Company/Team

Adobe/accessibility
